### PR TITLE
UICAL-145: Improvements to the Service Point selection on the Add Exception Modal

### PR DESCRIPTION
**Pull request by ncovercash**
## Purpose
Previously, the checkboxes to select the service point(s) for an exception had poor margins and looked strange.  This PR resolves these issues.

## Approach
All this does is resolve some margin and localization issues.

## Refs
https://issues.folio.org/projects/UICAL/issues/UICAL-145?filter=allopenissues

## Screenshots
Before:
image

After:
image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

**Comment by ncovercash** [comment]
Please note, I cannot request a review since I do not have any permissions here.

@Dmitriy-Litvinenko you're listed as the dev lead for this on [the matrix](https://wiki.folio.org/display/REL/Team+vs+module+responsibility+matrix) -- would you be the best person to review?  Or at least let me know who I should get to review and/or ask for access to request reviewers myself?

**Comment by dcrossleyau** [comment]
Hi Noah, thanks for your work with this and with the next ui-calendar PR.

The GitHub workflows from forks are new to us, so you are on the bleeding edge.

With [FOLIO-3290](https://issues.folio.org/browse/FOLIO-3290) we are attempting to address it.

@ankitasen-ubmainz  and @zburke -- I am having trouble understanding what the linked GitHub docs are saying.

Also having trouble with our new [documentation](https://dev.folio.org/guidelines/github-actions-frontend/#forked-repositories).

Would you please investigate.

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: Set styles directly to component - is not a good practice. Move them into separete .css file, and use classes. At string below it probably used because we get style directly from `item`.

File: src/settings/OpenExceptionalForm/ServicePointSelector.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ServicePointSelector.js:51]: Change constant string to translation is a good idea, but we should not change value of string. PO should decide to change it or not. So use please `No items to show` in translation file.

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: You should not change here any existed translations by two reasons:
- PO should deside it;
- this is not the goal of your ticket;

**Review by Stanislau-Snapok — COMMENTED** [review]
To commit every step what has been done - it's a good practice. But before merge `squash` them into one please. The final commit message  should looks like `UICAL-145: ...the problem it solves...`. Look at other commits in the repository please. Thanks for your work!

**Comment by ncovercash** [comment]
> To commit every step what has been done - it's a good practice. But before merge `squash` them into one please. The final commit message should looks like `UICAL-145: ...the problem it solves...`. Look at other commits in the repository please. Thanks for your work!

I thought this was done at merge-time, not as part of the PR?  If this is not the case I'll be happy to squash them.

@Stanislau-Snapok for the translation-related comments, would you recommend I make a separate PR for this?  While I was modifying the service point picker I saw that the label was reused and didn't apply nearly as well ("Valid To:") is a bit weird, especially since it was used on the field above it.

I'll resolve the CSS issue when I come into work next.

**Comment by Stanislau-Snapok** [comment]
> I thought this was done at merge-time, not as part of the PR? If this is not the case I'll be happy to squash them.
> 
> @Stanislau-Snapok for the translation-related comments, would you recommend I make a separate PR for this? While I was modifying the service point picker I saw that the label was reused and didn't apply nearly as well ("Valid To:") is a bit weird, especially since it was used on the field above it.
> 
> I'll resolve the CSS issue when I come into work next.

About `squash` - not on this project, i think. So I prefer do it localy.
About translations - yes, separate PR is a good idea. I think in case if we still have no ticket for this, we will be able to create it after your PR will be created, and in case if PO will agree with your changes.

**Comment by Dmitriy-Litvinenko** [comment]
> > To commit every step what has been done - it's a good practice. But before merge `squash` them into one please. The final commit message should looks like `UICAL-145: ...the problem it solves...`. Look at other commits in the repository please. Thanks for your work!
> 
> I thought this was done at merge-time, not as part of the PR? If this is not the case I'll be happy to squash them.
> 
> @Stanislau-Snapok for the translation-related comments, would you recommend I make a separate PR for this? While I was modifying the service point picker I saw that the label was reused and didn't apply nearly as well ("Valid To:") is a bit weird, especially since it was used on the field above it.
> 
> I'll resolve the CSS issue when I come into work next.

Hello, @ncovercash  Thank you. Could you please discuss in JIRA with PO should we change this translation or not.  I suggest ask PO to create new bug for change translation if we will need change them.

**Comment by ncovercash** [comment]
>About squash - not on this project, i think. So I prefer do it locally.

@Stanislau-Snapok it looks like GitHub has an option to merge and squash together (I cannot see it since I do not have write access): https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request screenshot

If I merge it locally, it would remove the entire point of commits (as all that would be visible on GH for review and/or documentation) is the squashed commit).  Seems counterproductive to remove one of the main features of git, but I could be wrong.

> Set styles directly to component - is not a good practice. Move them into separete .css file, and use classes. At string below it probably used because we get style directly from `item`.

I just fixed this in 6a167f5acd112003e765d09968aa53d412b6efda.  I also modified `CircleDiv` to `circle-div` to better match styles I found elsewhere in the project (there was a majority `dash-case`, a little `camelCase`, but no `PascalCase`).  The `!important` is required in order to override the `List` component's `justify-content: space-between` which would have higher precedence normally (and cause the issue that started this whole PR).

> Could you please discuss in JIRA with PO should we change this translation or not. I suggest ask PO to create new bug for change translation if we will need change them.

I have reverted the translation changes (I could not rebase as I had merged `master` since that commit) and opened [UICAL-177](https://issues.folio.org/projects/UICAL/issues/UICAL-177).

**Comment by Stanislau-Snapok** [comment]
@ncovercash, you right about `squash`, my fault =) Now you need approve from our team lead - @Dmitriy-Litvinenko. But I believe that now we can start pipeline. Be ready to fix some code style issues.

**Review by Stanislau-Snapok — APPROVED** [review]

**Comment by ncovercash** [comment]
Whoops, that test is definitely me – I'll see what I messed up on the localization reversion in the morning and fix that

**Comment by ncovercash** [comment]
Forgot to change the test label back -- fixed!

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: Change please attribute "class" to "className"
Also, to be honest, our team prefere use camelCase for class naming. So as we now responsible for this module, could you please use this style for all clasess that you change?
Thanks!

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: As you already change this line, could you please split properties into different lines. Also could you please apply this approach for all lines abowe and below, which you change and where number of properties is more than one.
Thanks!

File: src/css/exception-form.css
Comment (Stanislau-Snapok) [inline src/css/exception-form.css]: Add please space before {

File: CHANGELOG.md
Comment (Stanislau-Snapok) [inline CHANGELOG.md]: Remove please unnecessary second space after dot

**Review by Stanislau-Snapok — APPROVED** [review]

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (ncovercash) [reply src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: Sure!  I saw just about every casing style throughout the repo, so assumed dash-case incorrectly, and will fix this.

**Comment by ncovercash** [comment]
I've fixed the requested style issues -- if there is a style preference list for Vega, I would greatly appreciate a copy of it (could not find it when I searched).

**Comment by Stanislau-Snapok** [comment]
> I've fixed the requested style issues -- if there is a style preference list for Vega, I would greatly appreciate a copy of it (could not find it when I searched).

Don't see the latest changes. Only merge upstream into your repository. Could you please check your push one more time?

**Comment by ncovercash** [comment]
@Stanislau-Snapok my bad, I messed up the merge between my local and fork.  Should all be there now!

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: You should use className. Also ">" should be at the next line and be aligned with "<li"

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: move className at next line, please

File: src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ExceptionalPeriodEditor.js]: "/>" as with ">" above

File: src/settings/OpenExceptionalForm/ServicePointSelector.js
Comment (Stanislau-Snapok) [inline src/settings/OpenExceptionalForm/ServicePointSelector.js]: I checked your branch localy, and I think we can just remove this className. Then you can leave "style" as is, or wright it at the same line with the div. Since there is only one property here, both approaches are acceptable.

**Review by Stanislau-Snapok — COMMENTED** [review]
Thank you for your contibuting. I think this changes should be last. After them we need to resolve problem with pipeline fail. I read the doc, and it looks like one of us should make another PR with changes from your branch. The same issue with you PR for fix grammatical issues in template. We working on it, so I believe we can merge your changes soon.

File: src/settings/OpenExceptionalForm/ServicePointSelector.js
Comment (ncovercash) [reply src/settings/OpenExceptionalForm/ServicePointSelector.js]: I'm not sure why this got left as `circle-div`, it should have been made into `circleDiv`  but it does seem to be needed in order to make this div the proper height and circular dot.

**Comment by ncovercash** [comment]
@Stanislau-Snapok thank you so much for working with me on these style things, I'm sorry for how much this has been dragged out.  Working on learning the Vega styles/preferences (and hopefully this is the last time 🤞)

**Review by Stanislau-Snapok — APPROVED** [review]

**Comment by Dmitriy-Litvinenko** [comment]
Thank you @ncovercash for fix UICAL-145: The "Online" checkbox is not centered vertically.
We squashed all commits and create new pull https://github.com/folio-org/ui-calendar/pull/363 for resolve problem with merge.

**Pull request closed without merging**
